framework/src/main/java/org/checkerframework/framework/ajava/ExpectedTreesVisitor.java

```diff
@@ -0,0 +1,421 @@
+package org.checkerframework.framework.ajava;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.VarType;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import com.sun.source.tree.AnnotatedTypeTree;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.DoWhileLoopTree;
+import com.sun.source.tree.ExpressionStatementTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.ForLoopTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.IfTree;
+import com.sun.source.tree.ImportTree;
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.ModifiersTree;
+import com.sun.source.tree.NewArrayTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.SwitchTree;
+import com.sun.source.tree.SynchronizedTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.tree.WhileLoopTree;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.javacutil.BugInCF;
+
+/**
+ * Visitor that stores each visited tree that should match with some JavaParser node if the same
+ * Java file was parsed with both. The primary purpose is to test the {@link
+ * JointJavacJavaParserVisitor} class when the -AcheckJavaParserVisitor flag is used. That classes
+ * traverses a javac tree and JavaParser AST simultaneously, so the trees this class stores can be
+ * used to test if the entirety of the javac tree was visited.
+ *
+ * <p>Adds these trees to a set during traversal. Many trees shouldn't be matched with a JavaParser
```

Comment:
You don't need to list them here, but can you say how to find, in the source code below, all the trees that shouldn't be matched with a JavaParser node?  Is it just a search for the string "javac"?

Reply:
I don't quite understand what you mean. The class as a whole contains the instructions for how to find trees that shouldn't be matched with a JavaParser node. These exceptions occur at each call to `trees.remove` and when some visit method is overriden and a child is not visited. In that case the entirety of the child's subtree tree shouldn't be matched to JavaParser nodes. As far as I'm aware, all such exceptions are accompanied by a comment.